### PR TITLE
Updated callingSDK version to 2.1.0-beta.1 

### PR DESCRIPTION
**Pull request by JohnVUgonabo**
## Purpose

* Updated callingSDK version from 2.0.1-beta.1 to 2.1.0-beta.1
* Refactored codes to reflect newly introduced naming of classes and functions
* https://github.com/Azure/Communication/releases/tag/v2.1.0-beta.1

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: SDK Upgrade
```

## What to Check
Verify that the following are valid
* Verify that the Recording and Transcription notifications work as expected

## Other Information
SDK releases: https://github.com/Azure/Communication/releases
Task: https://skype.visualstudio.com/SPOOL/_sprints/backlog/ACS%20Mobile%20UI/SPOOL/CY2021-Q4/Sprint%2050%20(Ends%20November%2028)?workitem=2648038

### CODE REVIEW

File: AzureCalling/External/Calling/CallingContext.swift [hunk 1]
```diff
@@ -36,8 +36,8 @@ class CallingContext: NSObject {
     private var callClient: CallClient?
     private var callAgent: CallAgent?
     private var call: Call?
-    private var callRecordingFeature: RecordingFeature?
-    private var callTranscriptionFeature: TranscriptionFeature?
+    private var callRecordingFeature: RecordingCallFeature?
+    private var callTranscriptionFeature: TranscriptionCallFeature?
```
Comment (jimchou-dev) [inline AzureCalling/External/Calling/CallingContext.swift:40]: Maybe change the variable name to match the type name as well? to `recordingCallFeature` and `transcriptionCallFeature`

**Review by jimchou-dev — DISMISSED** [review]
Tested

File: AzureCalling/External/Calling/CallingContext.swift (hunk 1 above)
Comment (JohnVUgonabo) [reply AzureCalling/External/Calling/CallingContext.swift]: @jimchou-dev Thanks for pointing that out. PR has now been updated

File: AzureCalling/External/Calling/CallingContext.swift [hunk 2]
```diff
@@ -464,9 +464,8 @@ class CallingContext: NSObject {
     }
 }
 
-extension CallingContext: CallDelegate,
-                          RecordingFeatureDelegate,
-                          TranscriptionFeatureDelegate {
+extension CallingContext: CallDelegate, RecordingCallFeatureDelegate,
+                            TranscriptionCallFeatureDelegate {
```
Comment (cheukchuen) [inline AzureCalling/External/Calling/CallingContext.swift:468]: Nit: separate each delegates into a newline like before

**Review by JoshuaLai — DISMISSED** [review]

File: AzureCalling/External/Calling/CallingContext.swift (hunk 2 above)
Comment (JohnVUgonabo) [reply AzureCalling/External/Calling/CallingContext.swift]: @cheukchuen Thanks for the pointer, updated PR

**Review by JoshuaLai — APPROVED** [review]

**Review by cheukchuen — APPROVED** [review]

**Pull request merged**